### ISSUE TITLE
adding "fpixtest" TestType

### ISSUE THREAD

**Issue by lantone** [puhep/MoReWeb#1]
Hi all,

Could we add an entry for "fpixtest" to the available test types in the QualificationGroup, à la "purduetest"?:

https://github.com/puhep/MoReWeb/blob/pufpix/Analyse/TestResultClasses/CMSPixel/QualificationGroup/QualificationGroup.py#L50

We tried using the "automatic" test type but it's not really set up to parse data from an elcommandante session with multiple modules running in parallel.

Thanks,
Jamie

**Comment by lantone** [comment]
I just noticed that a block will also be needed in Controller.py.  Right now there's a command-line option, but it's not used by the code:

https://github.com/puhep/MoReWeb/blob/pufpix/Analyse/Controller.py#L25

Jamie

**Comment by khan62** [comment]
In the past we've had trouble with the command line options. Running without options is much more straightforward in the code, but it does have the quirk that it tries to run every folder in the data directory. However, I don't think we've ever run multiple modules in parallel, so I'm not sure how it acts or how to fix it. Would you be able to send me to data you're trying to parse so I can try it on my machine?

**Comment by khan62** [comment]
I was also reading through the email thread, and I think we made a small error. We have the  GlobalOverviewPath and GlobalFinalResultsPath set as the same location, not the GlobalDataDirectory and the GlobalOverviewPath. 

Here's ours, for example:
GlobalDataDirectory = /project/cmsfpix/.www/Submission_p/morewebInput

GlobalOverviewPath = /project/cmsfpix/.www/MoReWeb/Results

GlobalFinalResultsPath = /project/cmsfpix/.www/MoReWeb/Results

**Comment by lantone** [comment]
Hi Kamal,

I copied an output folder to dropbox:

https://www.dropbox.com/sh/rd9i1u3oilyw3s3/AACBGQkLmwTI4oiphG9Ky43Ha?dl=0

Cheers,
Jamie

**Comment by khan62** [comment]
I know MoReWeb recurses through the directories based on the numerical prefix. I expect your issue has something to do with the fact that your directories jump from 000 to 004. Did pxar not output any 001, 002, or 003 directories?

**Comment by lantone** [comment]
Yeah, I'm pretty sure that's what's going wrong.  We test 4 modules in series, so their full test results get output test directories labeled 001, 002, 003, and 004 respectively.  That's why I was hoping to use the command-line option to just go get the results of the FPixTest directly, since the automated algorithm fails in our case.  I take it your implementation of "purduetest" not work then?

**Comment by khan62** [comment]
I changed the IV directory to "005_IV_p17" and I was able to run your data:
http://www.physics.purdue.edu/cmsfpix/MoReWeb/Results/REV001/R001/P-A-3-22_Elcomandantetest_2015-06-22_15h58m_1435006731/QualificationGroup/ModuleFPIXTest_p17_1/TestResult.html

(I wasn't expecting it to be as successful as it was)

Is your other data set up in similar directories? If so, it seems that moving the IV directory to come after the data directory may fix the problem. Also, please try running this fix on your machine to make sure it's replicable.

**Comment by lantone** [comment]
Hi Kamal,

Great, good to know that the rest of the workflow functions once we get past this issue.  I'm a bit nervous to go with a solution that you didn't expect to work in the first place...  Reordering the tests might end up working for this use case, but it doesn't seem like a very robust solution.  Maybe @jstupak can comment?

Jamie

**Comment by jstupak** [comment]
I second Jamie's opinion on robustness, but this might be an okay solution in the short term.  There is really no reason we have to run the IV before the FPIXTest, so I will try running the tests in the opposite order today, and see if everything works.  We can go from there

**Comment by khan62** [comment]
I realized that, though the MoReWeb ran without throwing any errors, it was not including the IV scan. After some experimenting and sifting through the code, I found a way for the IV scan to be included:
- In configfiles/elcomandante.ini, there is a section called "[Tests]". Here there is a value called "Test", which is a comma-delimited list. I moved "FPIXTest@17" from the end to the front of this list.
- I renamed "004_FPIXTest_p17" to "000_FPIXTest_p17"
- I renamed "005_IV_p17" to "001_IV_0_p17"

I think what is happening is that MoReWeb reads elcomandante.ini to find the order of the tests, then cycles through those and parses the tests in that order. However, it only parses IV data if it is right after another test. Because FPIXTest was at the end, none of the IV data was accessed. By changing these values, MoReWeb reads FPIXTest first, then accesses the first IV scan after it.

**Comment by gfunk723** [comment]
We also have an issue with the powercycle step not incrementing the label number so there is a 000_powercycle directory and a 000_FPIXTest_p17 directory. 

If the powercycle step is removed from configfiles/elcomandante.ini, MoReWeb begins to run, but with many errors. It is looking for SCurveData_C*.dat files that are not in our output, and some other object attributes that don't exist. It creates an html, but with most of the data missing. 

The full output log from running Controller.py is here: https://gist.github.com/gfunk723/8677f0cd2c0da4473a67

**Comment by jstupak** [comment]
And the award for best github username goes to... Garret :)

I think this latest problem is my fault.  I changed around a bit the way we perform the tests, and it looks like the SCurvesData files are no longer saved.  I know we used to have them, so Ill assume they were in the example tarball I have Kamal before.  @gfunk723 can you please make a list of all the files moreweb complains are missing, so we can figure out what I need to make sure we keep.

**Comment by gfunk723** [comment]
We've made some progress and eliminated a lot of the errors from before. The output html is only missing several test results now, and one of them is the Untrimmed Vcal Threshold test. Is it possible to get MoRe-Web to ignore this and not try to include it in the output? It is not necessary to display this result, and it would simplify finding the rest of the problems.
